Dr-Nekoma/RacketowerDB
# A farewell to the empirically proven to be failed modern OOP

# Goal
Once upon a time, there were a @EduardoLR10 and a @MMagueta that wanted to go against their own bias towards functional programming and immutability. They tried very hard to embrace a modern approach to separation of concerns using modern, a.k.a, post-Java, object-oriented programming for RacketowerDB.
They managed to go far but eventually, reality hit, and the OOP monument crumbled with their dreams inside. Confident that it was not a matter of bias, they were enlightened by @z-silver to change the design of the application to use a combination of immutable structs, generic interfaces, and contracts. Such a set of abstractions proved to be a far better alternative in comparison, and the threefold team dedicated a significant amount of time to making an overall design change that would allow them to dream again.

# Related Issues

Close #1.

# How to Test

- Make sure you delete all data files, i.e., within the `ndf/data` folder.
- Run `nix develop && just build && ./main` and you **should not** see any errors.

# Observations for the Reviewer

I hope you understand that this is a small step for us, but one giant leap for the future of the official database project of DrN.
